xmodule/xml_block.py

```diff
@@ -498,6 +498,11 @@ def add_xml_to_node(self, node):
             node.set('org', self.location.org)
             node.set('course', self.location.course)
 
+        # Serialize and add tag data if any
+        tag_data = self.serialize_tag_data()
+        if tag_data:
+            node.set('tags-v1', tag_data)
```

Comment:
```suggestion
        if isinstance(self, TaggedBlockMixin):
            tag_data = self.serialize_tag_data()
            if tag_data:
                node.set('tags-v1', tag_data)
```

I suggest you do this, then you don't have to update the LMS or all those test cases.

Reply:
Update to this, it works well, thanks!